docs/faq.md

```diff
@@ -280,3 +280,27 @@ Note: Windows with Radeon GPUs currently default to 1 model maximum due to limit
 ## How does Ollama load models on multiple GPUs?
 
 Installing multiple GPUs of the same brand can be a great way to increase your available VRAM to load larger models.  When you load a new model, Ollama evaluates the required VRAM for the model against what is currently available.  If the model will entirely fit on any single GPU, Ollama will load the model on that GPU.  This typically provides the best performance as it reduces the amount of data transfering across the PCI bus during inference.  If the model does not fit entirely on one GPU, then it will be spread across all the available GPUs.
+
+## How can I set the quantization type for the K/V cache?
+
+The K/V context cache can be quantized to significantly reduce memory usage when Flash Attention is enabled.
```

Comment:
Does the KV cache only work with Flash Attention? Or in all cases?

Reply:
No. Here is one example that requires being able to specify these individually:
DeepSeek - can not use flash attention, can only quantize the K cache, cannot quantize the V cache

Reply:
Thanks!

Is that encoded anywhere in the model config/metadata? That way we could catch these edge cases at runtime vs erroring

Reply:
Great question. I really don't know how this is being checked inside llama.cpp. It's likely a current limitation of llama.cpp rather than an issue with the model itself.

Reply:
Do you think it would somehow be possible to set an automatic fallback where k or v cache falls back to f16 in the event of an error? 

Then it would just automatically switch k, v or both back to f16 when incompatible. 

That way it would just add a few seconds to load time while it reloads models properly so user wouldn't have to deal with errors or settings in these cases.

Reply:
Not my software, but I wouldn't overcomplicate things. If the desired parameters aren't correct and the model fails to load I'd error out and fail to load the model. Here are a few reasons:

- Adding workaround code adds complexity
- The code may need to be changed if llama.cpp (or another) back end changes their ability to load the model under these conditions
- It may give the user the improper impression that the settings are working even though they were silently changed during runtime

If someone is advanced enough to be messing with these settings they should validate that they work correctly. That's even the case with "simple" settings like context length.

Reply:
I think we could see that easily in logs if it was indeed functioning as intended, it would make it easier for end users and advanced users could easily double check it. 

But in the end just a suggestion, definitely not my software either. Haha

Reply:
Only with flash attention.

Reply:
Honestly I don’t understand why anyone would disable Flash Attention, I know a few older architecture models don’t support it but really it should always be enabled for models that do.

Reply:
Awesome, thanks @sammcj ! Completely agree. Would love to turn it on by default (probably in a separate PR)! Only time we'd disable it would be for compatibility or if its unsupported.. do you know which models might (we can bring this to another issue/PR) support it by default?

Reply:
The only model I've personally encountered in recent times that don't support FA is deepseek-coder-v2(-lite), but the behaviour in Ollama as it is means if you have FA enabled and you run an older non-FA model - it will just disable FA which is nice behaviour.

I agree this could be done on another issue/PR, it's something that would greatly benifit folks that "just use" Ollama without tuning etc... as they would need far less vRAM to run any decent context size.